### ISSUE TITLE
THUMBNAIL_NAMER setting does not have any effect when using filer

### ISSUE THREAD

**Issue by benzkji** [django-cms/django-filer#1093]
When dealing with SEO (https://github.com/SmileyChris/easy-thumbnails/issues/474) or when one just wants a different THUMBNAIL_NAMER (https://github.com/SmileyChris/easy-thumbnails/issues/492), there seems to be no luck with the filer.

There is the `ThumbnailerNameMixin`, that has it's own `get_thumbnail_name()` method, where the naming is fixed, as said there "A version of ``Thumbnailer.get_thumbnail_name`` that produces a
        reproducible thumbnail name that can be converted back to the original
        filename. : https://github.com/divio/django-filer/blob/006a1cfbdb5f61b6725b7675c7f9117e85b5d367/filer/utils/filer_easy_thumbnails.py#L32 That mixin is used for `FilerThumbnailer`, that is then created here: https://github.com/divio/django-filer/blob/006a1cfbdb5f61b6725b7675c7f9117e85b5d367/filer/models/abstract.py#L175 . I do not really get how this all works and interacts with easy_thumbnails. But the question remains: Why does filer need to override the namer, and get back to the original from the thumbnail? Normally, it would be enough to get the thumbnails, when for example deleting a file, so no orphaned thumbs are around?

And finally, am I missing something in the docs, and setting a custom namer is already possible?

**Comment by benzkji** [comment]
Ah. I guess getting to the original is needed when verifying access thumbnails of secured files/images?! Making a custom namer possible on non-protected files would be a nice one, then :)

**Comment by benzkji** [comment]
a quick feedback from the team would be nice, like "we wont ever support" or "sure, wait, we'll do that right now, it's so easy".

thanks in advance @yakky @stefanfoulis ?

**Comment by ianare** [comment]
In order to work around this issue I've configured my nginx server to redirect all resized images to the original, but only on search engine user agent.

In this way Google always gets the full sized image, and I don't have to worry about changing image dimensions in thumbnails.

Not saying this is ideal... but seems to work well enough.

**Comment by benzkji** [comment]
thanks for the insight. least but ideal - you'll get really bad scores in page speed insights, this way...this being only the first thing coming to my mind ...

**Comment by ianare** [comment]
I'm getting good scores on page speed insights (96 on dekstop / 54 on mobile), the bad score on mobile is due to "Defer offscreen images" and "Eliminate render-blocking resources" which we are in the process of fixing. PageSpeed uses a mobile Chrome user agant so is not impacted by this.

Here is my nginx mapping if of some interest to others:
```
#
# Search engines
#
map $http_user_agent $search_engine_agent {
    default 0;

    "~Applebot" 1;
    "~AhrefsBot" 1;
    "~Google" 1;
    "~[Bb]ingbot" 1;
    "~Slurp" 1;
    "~Crawlera" 1;
    "~Curious George" 1;
    "~LinkedInBot" 1;
    "~DuckDuckBot" 1;
    "~Baiduspider" 1;
    "~YandexBot" 1;
    "~LinkedInBot" 1;
    "~Sogou" 1;
    "~[Ee]xabot" 1;
    "~facebot" 1;
    "~ia_archiver" 1;
    "~Gigabot" 1;
    "~Gigablast" 1;
    "~Twitterbot" 1;
    "~OrangeBot" 1;
    "~MJ12bot" 1;
    "~SemrushBot" 1;
    "~SkypeUriPreview" 1;
}
```
and then it's used as follows in the virtual host conf file:
```
# search engines always get the high-resolution image
if ($search_engine_agent) {
    rewrite ^/media/filer_public_thumbnails/filer_public/(.+)__.+$ /media/filer_public/$1 permanent;
}
```

**Comment by benzkji** [comment]
how about digging this one out - @jrief ? I hate monkey patching ;-)

**Comment by benzkji** [comment]
for me, this is an important issue. please re-open. I would even think about creating a PR, since they are reviewed more efficiently now...

**Comment by fsbraun** [comment]
@benzkji Thanks for keeping this alive! It would be wonderful if you could create a PR!

**Comment by benzkji** [comment]
@fsbraun yes, a PR would be golden. Maybe I'll try. Just remove the fields, or put them on models.SET_NULL? v4 only still holds, is that correct? I'm not on v4 with any of my projects, migrations and non-versioning did not have priority, here...|

**Comment by fsbraun** [comment]
@benzkji Are you talking about https://github.com/django-cms/django-cms/pull/8167 ?

Looked into this very issue and realized that the fixed file name convention is part of filers performance optimization. I can think of a solution which allows overwriting `THUMBNAIL_NAMER` with the drawback of a slower folder list view (if overwritten).

**Comment by benzkji** [comment]
sorry for the confusion! I would really appreaciate a solution here, it bugged me for quite a while, but again it has (even very) low priority for me, now. 

But, it would help for SEO, and could be a really nice feature. I thinks that many people don't even really know about the protected files in filer ;)

**Comment by fsbraun** [comment]
@benzkji Can you test run @earthcomfy's patch here: #1519 
I'll give it a go, too.
